docs/source/features/parallelisms.rst

```diff
@@ -1,56 +1,48 @@
-distributed between GPUs, communication is required in order to keep
-the model copies consistent with each other.
+Data Parallelism (DP) replicates the model across multiple GPUs.
+Data batches are evenly distributed between GPUs and the data-parallel GPUs process them independently.
+While the computation workload is efficiently distributed across GPUs, inter-GPU communication is required in order to keep the model replicas consistent between training steps.
 
 Distributed Data Parallelism
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Distributed Data Parallelism (DDP) keeps model copies consistent by
-synchronizing parameter gradients before each optimization step. More
-specifically, it sums gradients over all model copies using an
-all-reduce communication collective.
+Distributed Data Parallelism (DDP) keeps the model copies consistent by synchronizing parameter gradients across data-parallel GPUs before each parameter update.
+More specifically, it sums the gradients of all model copies using all-reduce communication collectives.
 
 .. image:: ../nlp/nemo_megatron/images/ddp.gif
     :align: center
     :width: 800px
     :alt: Distributed Data Parallel
 
-Distributed Optimizer (ZeRO-1)
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Distributed Optimizer
+^^^^^^^^^^^^^^^^^^^^^
 
-The ZeRO-1 algorithm keeps model copies consistent by sharding the
-optimizer state between GPUs. During each optimization step, the
-parameter gradients are first summed and sharded (with a
-reduce-scatter collective), each GPU applies an optimization to its
-local shard of the parameters, and the updated parameter shards are
-broadcast to update all of the model copies (with an all-gather
-collective). This approach is attractive for large models since
-sharding the optimizer state can significantly reduce its memory
-footprint on individual GPUs. It also has, in theory, the same
-communication volume as DDP and its communication pattern has more
-opportunities for overlapping with compute.
+Distributed optimizer is a memory-optimized data-parallel deployment method.
+It shards the optimizer states and the high-precision master parameters across data-parallel GPUs instead replicating them.
+At the parameter optimizer step, each data-parallel GPU updates its shard of parameters.
+Since each GPU needs its own gradient shard, the distributed optimizer conducts reduce-scatter of the parameter gradients instead of all-reduce of them.
+Then, the updated parameter shards are all-gathered across data-parallel GPUs.
+This approach significantly reduces the memory need of large scale LLM training.
+Also, when the precision of the gradient is higher than the parameter precision, the split execution of gradient reduce-scatter and parameter all-gather can reduce the total communication volume.
+This split collective execution increases the total computation to overlap with the communication, which improves the overlap opportunity.
```

Comment:
By splitting the collective execution, you increase the total computation to overlap with communication, thereby improving the opportunity for overlap.